usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -489,6 +519,467 @@ static CK_RV cleanse_attribute(TEMPLATE *template,
     return CKR_OK;
 }
 
+/*******************************************************************************
+ *
+ *                    Begin EP11 protected key option
+ */
+
+typedef struct {
+    CK_BBOOL wrap_was_successful;
+    CK_VOID_PTR secure_key;
+    CK_ULONG secure_key_len;
+    CK_BYTE *pkey_buf;
+    size_t *pkey_buflen_p;
+} pkey_wrap_handler_data_t;
+
+/* A wrapped key can have max 132 bytes for an EC-p521 key */
+#define MAX_WRAPPED_KEY_SIZE    (2 * (521 / 8 + 1))
+typedef struct {
+    uint16_t version;
+    uint8_t res0[16];
+    uint32_t cpacf_key_type;
+    uint32_t bit_length;
+    uint64_t token_size;
+    uint8_t res1[8];
+    uint8_t wrapped_key[MAX_WRAPPED_KEY_SIZE];
+    uint8_t res2[50];
+} __attribute__((packed)) wrapped_key_t;
+
+/**
+ * Callback function used by handle_all_ep11_cards() for creating a protected
+ * key via the given APQN (adaper,domain).
+ */
+static CK_RV ep11tok_pkey_wrap_handler(uint_32 adapter, uint_32 domain,
+                                       void *handler_data)
+{
+    CK_BYTE iv[16] = {
```

Comment:
const ?

Reply:
Yes, I already tried it, but it results in ugly casts when using the byte array later

Reply:
I think unfortunately the pkcs11 types fail at constifying things, so if thats the case here, id rather leave it as it ist. Theres not sense in putting a const and later forcing a cast.